jxline-core/src/main/java/cloud/xline/jxline/impl/ClientImpl.java

```diff
@@ -3,52 +3,33 @@
 import cloud.xline.jxline.Client;
 import cloud.xline.jxline.ClientBuilder;
 
-import io.etcd.jetcd.*;
+import cloud.xline.jxline.KV;
+import cloud.xline.jxline.ProtocolClient;
+import io.etcd.jetcd.support.MemorizingClientSupplier;
 
 public final class ClientImpl implements Client {
 
-    public ClientImpl(ClientBuilder clientBuilder) {}
+    private final ClientConnectionManager manager;
 
-    @Override
-    public Auth getAuthClient() {
-        return null;
-    }
-
-    @Override
-    public KV getKVClient() {
-        return null;
-    }
-
-    @Override
-    public Cluster getClusterClient() {
-        return null;
-    }
+    private final ProtocolClient protocolClient;
 
-    @Override
-    public Maintenance getMaintenanceClient() {
-        return null;
-    }
+    private final MemorizingClientSupplier<KV> kvClient;
 
-    @Override
-    public Lease getLeaseClient() {
-        return null;
+    public ClientImpl(ClientBuilder clientBuilder) {
+        this.manager = new ClientConnectionManager(clientBuilder);
+        this.protocolClient = new ProtocolClientImpl(this.manager);
+        this.kvClient =
+                new MemorizingClientSupplier<>(
```

Comment:
Is this wrapper for lazy initialization?

Reply:
yes